### ISSUE TITLE
:LspStatus returns typescript-language-server: not running

### ISSUE THREAD

**Issue by OurCodeBase** [mattn/vim-lsp-settings#743]
Why `:LspStatus` returns `typescript-language-server:` not running ?
Please show me how to run `typescript-language-server` in neovim.
```
$ vi helloworld.js
:PlugInstall
:LspInstallServer
:LspStatus
typescript-language-server: not running
```

I have read closed issue #547 and included these lines to `init.vim` file:-
```
let g:lsp_diagnostics_echo_cursor = 1
let g:lsp_settings = {
\ 'efm-langserver': {
\   'disabled': v:false
\ },
\ }

autocmd BufWritePre *.js,*.jsx call execute('LspDocumentFormatSync --server=efm-langserver')
```
But the error still appears the same...

**Comment by mattn** [comment]
Question:

* Do it have deno.json exists on root of your project directory?
* Do it have node_modules on root of your project directory?

**Comment by OurCodeBase** [comment]
No, I'm using `tsserver` on a standalone `script.js` file. Expecting auto-completions on atleast `console.log()`.

**Comment by subeenregmi** [comment]
Creating a folder `node_modules` in the same directory of my `script.js` file seemed to make the language server work for me.

**Comment by cheungxi** [comment]
have same question in vim9

**Comment by mattn** [comment]
If you want to specify language server, set 
```
let g:lsp_settings_filetype_javascript = ['typescript-language-server']
```

**Comment by TayTTo** [comment]
@mattn there are serveral lsp for javascript, which is the best, and why my biome running but I can't use it for formating using LspDocumentFormat command

**Comment by mattn** [comment]
Sorry, I'm not familier to LSP for JS.

**Comment by TayTTo** [comment]
@mattn tks

**Comment by TayTTo** [comment]
@mattn hey bro, when I install tailwindcss-intellise, it not work when I want to use, is there any guide or special configuration for be able to use it?

**Comment by mattn** [comment]
You can use it for svelte filetype.

**Comment by subeenregmi** [comment]
After trying to figure this out again on a new laptop, I think I have found the cause. `.tsx` files have a filetype of `typescript.tsx`. By setting the filetype of `.tsx` files to `typescriptreact` the LSP worked.

Adding this my `.vimrc` fixed this issue:
```
autocmd BufRead,BufNewFile *.tsx set filetype=typescriptreact
```
